js-core/homeworks/homework-4/src/main.js

```diff
@@ -0,0 +1,147 @@
+/*
+* Напишите функцию которая будет принимать 1 аргумент и в зависимости от типа аргумента
+*
+* Если тип аргумента число или объект -> возвращать true
+* Если тип аргумента функция -> возвращать false
+* Если тип аргумента Строка и длина этой строки не равна 10 -> возвращать "длина вашей строки: <длина строки>
+*                            Если длина 10 -> 'you win'
+*
+*
+*
+*
+* */
+
+function typeOfArgument(data){
+	if(!isNaN(+data) || (!Array.isArray(data) && typeof data === 'object')){
+		return true;
+	}else if(typeof data === 'function'){
+		return false;
+	}else if(typeof data === 'string'){
+		return data.length!=10 ? `длина вашей строки: ${data.length}`:`you win`;
+	}
+}
+
+const func = function foo(){};
+const str = `asdfasdfasd`;
+const num = 11;
+const obj = {param1:1,param2:2};
+
+console.log(typeOfArgument(obj));
+
+/*
+*
+*  Задача в классе !
+*
+* */
+
+
+/*
+ 1. Напишите функцию которая принимает 2 числа
+ и возвращает массив содержащий числа между первым числом и вторым числом;
+ */
+
+function numbersBetween(a, b) {
+	const min = Math.min(a,b);
+	const max = Math.max(a,b);
+	let arr = [];
+	let count = 0;
+	for(let i = min;i<=max;i++){
+		arr[count]=i;
+		count++;
+	}
+	return arr;
+}
+
+console.log(numbersBetween(3, 5));
+// 3, 4, 5
+
+console.log(numbersBetween(10, 12));
+// 10, 11, 12
+
+
+/*
+ 2. Перепишите задачу FizzBuzz, но с использованием цикла.
+ Расчет чисел должен идти до 100
+ */
+
+// 1. FizzBuzz 3, 5, 3 && % 5
+
+function FizzBuzz(number) {
+		let result = ``;
+		if(number%3!=0 && number%5!=0){
+			result = number;
```

Comment:
such indention seems a bit weird in javascript world :)
please try to use 2 spaces